source/common/upstream/outlier_detection_impl.cc

```diff
@@ -477,10 +477,11 @@ void DetectorImpl::ejectHost(HostSharedPtr host,
                              envoy::data::cluster::v3::OutlierEjectionType type) {
   uint64_t max_ejection_percent = std::min<uint64_t>(
       100, runtime_.snapshot().getInteger(MaxEjectionPercentRuntime, config_.maxEjectionPercent()));
-  double ejected_percent = 100.0 * ejections_active_helper_.value() / host_monitors_.size();
+  double ejected_percent = 100.0 * (ejections_active_helper_.value() + 1) / host_monitors_.size();
```

Comment:
Ejected_percent should indicate the percentage **_after_** the ejection for max_ejection_percent to be applied correctly.

The issue is explained in detail here: https://github.com/envoyproxy/envoy/issues/27066